src/lib/bcmp/bm_rtos/bm_rtos.h

```diff
@@ -0,0 +1,67 @@
+#ifndef BM_RTOS_H
+#define BM_RTOS_H
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// base it off the posix error codes?
+typedef enum BmError {
+  BM_SUCCESS = 0,
+  BM_FAIL = -1,
+} BmError;
+
+// Something like this for the asserts? Although maybe this shouldn't live here
+// #ifdef DEV_MODE
+// #define ASSERT(x) configASSERT(x)
+// #else
+// #define ASSERT(x) ((void)0)
+// #endif
+
+typedef void *BmQueue;
+typedef void *BmSemaphore;
+typedef void *BmTimer;
+
+// Memory functions - these may not necessarily be tied to the RTOS but I'm including them here for now
+void *bm_malloc(size_t size);
+void bm_free(void *ptr);
+
+// Queue functions
+BmQueue bm_queue_create(uint32_t queue_length, uint32_t item_size);
+BmError bm_queue_receive(BmQueue queue, void *item, uint32_t timeout_ms);
+BmError bm_queue_send(BmQueue queue, const void *item, uint32_t timeout_ms);
+
+// Semaphore functions
+BmSemaphore bm_semaphore_create(void);
+BmError bm_semaphore_give(BmSemaphore semaphore);
+BmError bm_semaphore_take(BmSemaphore semaphore, uint32_t timeout_ms);
```

Comment:
Where is this defined at?

Reply:
Are you talking about what we talked about wrg to the:
```
#define MUTEX_UNLOCK (timeout) mutext_lock(timeout) {

#define MUTEX_LOCK mutex_lock()}
```
?

Reply:
No I do not see it's definition in the source file.

Reply:
woops! good eye!